homeassistant/bootstrap.py

```diff
@@ -93,8 +95,30 @@
         finally:
             clear_secret_cache()
 
-    if safe_mode or config_dict is None or not basic_setup_success:
+    if config_dict is None:
+        safe_mode = True
+
+    elif not basic_setup_success:
+        _LOGGER.warning("Unable to set up core integrations. Activating safe mode")
+        safe_mode = True
+
+    elif "frontend" not in hass.config.components:
+        _LOGGER.warning("Detected that frontend did not load. Activating safe mode")
```

Comment:
How do we know that the user requested the frontend to load?

Reply:
We don't. But I don't know of any use case that runs without the frontend loaded.

Reply:
We haven't required the frontend to be loaded before this PR, as far as I know.

If you're relying mainly on automations the frontend is mostly used for configuration, and that can be done without frontend, I think. So there definitely are use cases.

I think this is a pretty big change to do without discussion 1 day before a beta cut.

Reply:
https://github.com/home-assistant/home-assistant/pull/31969